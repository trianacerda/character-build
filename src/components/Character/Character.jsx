export default function Character({ head, middle, bottom }) {
  return (
    <section style={{ display: 'flex', flexDirection: 'column' }}>
      <article
        className="head"
        style={{
          backgroundImage: `url(./${head}-head.png)`,
          height: 200,
          width: 200,
          backgroundSize: 'cover',
          backgroundRepeat: 'no-repeat',
        }}
      ></article>

      <article
        className="middle"
        style={{
          backgroundImage: `url(./${middle}-middle.png)`,
          height: 200,
          width: 200,
          backgroundSize: 'cover',
          backgroundRepeat: 'no-repeat',
        }}
      ></article>

      <article
        className="bottom"
        style={{
          backgroundImage: `url(./${bottom}-bottom.png)`,
          height: 200,
          width: 200,
          backgroundSize: 'cover',
          backgroundRepeat: 'no-repeat',
        }}
      ></article>
    </section>
  );
}
