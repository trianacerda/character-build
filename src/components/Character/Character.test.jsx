import { screen, render } from '@testing-library/react';
import Character from './Character';

it('should render Character element on page', async () => {
  const { container } = render(<Character head="beth" middle="cowboy" bottom="gummybear" />);
  const element = await screen.findAllByRole('article');

  expect(element[0]).toBeInTheDocument();
  expect(container).toMatchSnapshot();
});
