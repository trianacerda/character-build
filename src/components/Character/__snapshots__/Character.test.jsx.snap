// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Character element on page 1`] = `
<div>
  <section
    style="display: flex; flex-direction: column;"
  >
    <article
      class="head"
      style="background-image: url(./beth-head.png); height: 200px; width: 200px; background-size: cover; background-repeat: no-repeat;"
    />
    <article
      class="middle"
      style="background-image: url(./cowboy-middle.png); height: 200px; width: 200px; background-size: cover; background-repeat: no-repeat;"
    />
    <article
      class="bottom"
      style="background-image: url(./gummybear-bottom.png); height: 200px; width: 200px; background-size: cover; background-repeat: no-repeat;"
    />
  </section>
</div>
`;
